DUI3-DX/Sdk/Speckle.Converters.Common.DependencyInjection/ToHost/ToHostConverterWithFallback.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Speckle.Autofac.DependencyInjection;
+using Speckle.Converters.Common.Objects;
+using Speckle.Core.Models;
+using Speckle.Core.Models.Extensions;
+
+namespace Speckle.Converters.Common.DependencyInjection.ToHost;
+
+public sealed class ToHostConverterWithFallback : ISpeckleConverterToHost
+{
+  private readonly IFactory<string, ISpeckleObjectToHostConversion> _toHost;
+  private readonly ToHostConverterWithoutFallback _baseConverter;
+
+  public ToHostConverterWithFallback(IFactory<string, ISpeckleObjectToHostConversion> toHost)
+  {
+    _toHost = toHost;
+    _baseConverter = new ToHostConverterWithoutFallback(toHost);
+  }
+
+  /// <summary>
+  /// Converts a <see cref="Base"/> instance to a host object.
+  /// </summary>
+  /// <param name="target">The <see cref="Base"/> instance to convert.</param>
+  /// <returns>The converted host object.
+  /// Fallbacks to display value if a direct conversion is not possible.</returns>
+  /// <remarks>
+  /// The conversion is done in the following order of preference:
+  /// 1. Direct conversion using the <see cref="Speckle.Converters.Common.DependencyInjection.ToHost.ToHostConverterWithoutFallback.TryConvert(Base, out object?)"/> method.
+  /// 2. Fallback to display value using the <see cref="Speckle.Core.Models.Extensions.BaseExtensions.TryGetDisplayValue{T}"/> method, if a direct conversion is not possible.
+  ///
+  /// If the direct conversion is not available and there is no displayValue, a <see cref="System.NotSupportedException"/> is thrown.
+  /// </remarks>
+  /// <exception cref="System.NotSupportedException">Thrown when no conversion is found for <paramref name="target"/>.</exception>
```

Comment:
Seems like also throws `InvalidOperationException` via private `FallbackToDisplayValue` function

Reply:
It does, but that is not the normal execution flow, and devs should not prepare against it (as they cannot recover from it)

If a connector doesn't support fallbacks it shouldn't use the fallback default converter 😁

Reply:
It does, but we don't want callers to handle the `InvalidOperationException`, if a connector registeres his `ToHostConverterWithFallback` but forgets to register a DisplayValue converter...

I'm ok about being a bit selective with which exceptional paths are documented, as we want to craft a clear DX for users of this function, and this failure case is not one.

Something we would want to fail loudly. If a connector doesn't need fall-back, we have the `ToHostconvererWithoutFallback`